api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -58,9 +58,16 @@ private Types() {}
           .put(BinaryType.get().toString(), BinaryType.get())
           .put(UnknownType.get().toString(), UnknownType.get())
           .put(VariantType.get().toString(), VariantType.get())
+          .put(GeometryType.crs84().toString(), GeometryType.crs84())
+          .put(GeographyType.crs84().toString(), GeographyType.crs84())
           .buildOrThrow();
 
   private static final Pattern FIXED = Pattern.compile("fixed\\[\\s*(\\d+)\\s*\\]");
+  private static final Pattern GEOMETRY_PARAMETERS =
+      Pattern.compile("geometry\\s*(?:\\(\\s*([^,]+?)\\s*\\))?", Pattern.CASE_INSENSITIVE);
```

Comment:
I typically like to match on any character other than the next delimiter, which in this case is `)`. Rather than matching any non-comma character with `([^,]+?)`, using `([^)]+?)` instead will cover more cases.

We don't want a string to fail matching the regex but that's what would happen in cases like `geometry(srid:123,456)`. Instead of failing to match, I think it will be better to parse correctly and then state that the CRS is unknown.

Reply:
Use `([^)]+?)` to capture more cases and check for `,` after matching.